File: api-tester/src/main/java/com/revenuecat/apitester/java/NoCoreLibraryDesugaringExceptionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.NoCoreLibraryDesugaringException;

@SuppressWarnings({"unused"})
final class NoCoreLibraryDesugaringExceptionAPI {
    static void check() {
        final Throwable cause = new Throwable();
        throw new NoCoreLibraryDesugaringException(cause);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/NoCoreLibraryDesugaringExceptionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.NoCoreLibraryDesugaringException

@Suppress("unused", "UNUSED_VARIABLE", "ThrowingExceptionsWithoutMessageOrCause")
private class NoCoreLibraryDesugaringExceptionAPI {
    fun check() {
        val cause = Throwable()
        val exception = NoCoreLibraryDesugaringException(cause)
        val runtimeException: RuntimeException = exception
        throw exception
    }
}
